tests/integration/targets/aci_outofsvc_interface_blacklist/tasks/main.yml

```diff
@@ -0,0 +1,357 @@
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: debug
+    test_vars_leaf: &test_vars_leaf
+      pod_id: 1
+      node_id: 1101
+      interface: eth1/33
+    test_vars_spine: &test_vars_spine
+      pod_id: 1
+      node_id: 1201
+      interface: eth1/33
+    test_vars_fex: &test_vars_fex
+      pod_id: 1
+      node_id: 1101
+      interface: eth123/1/33
+
+- name: Query system information
+  aci_system:
+    <<: *aci_info
+    id: 1
+    state: query
+  register: version
+
+- name: Leaf - Clean test environment with enabled interface
+  cisco.aci.aci_outofsvc_interface_blacklist:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: absent
+
+- name: Leaf - Query outofsvc_interface_blacklist
+  cisco.aci.aci_outofsvc_interface_blacklist:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: query
+  register: query_int_enabled
+
+- name: Leaf - Verify that outofsvc_interface_blacklist is not present
+  assert:
+    that:
+    - query_int_enabled.current == []
+
+- name: Leaf - Disable interface in check mode
+  cisco.aci.aci_outofsvc_interface_blacklist:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: present
+  check_mode: yes
+  register: disable_int_check
+
+- name: Leaf - Verify that outofsvc_interface_blacklist is not created after check mode
+  assert:
+    that:
+    - disable_int_check.current == []
+
+- name: Leaf - Disable interface
+  cisco.aci.aci_outofsvc_interface_blacklist:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: present
+  register: disable_int
+
+- name: Leaf - Verify that outofsvc_interface_blacklist is created
+  assert:
+    that:
+    - disable_int.current.0.fabricRsOosPath.attributes.dn == "uni/fabric/outofsvc/rsoosPath-[topology/pod-1/paths-1101/pathep-[eth1/33]]"
+    - disable_int.current.0.fabricRsOosPath.attributes.lc == "blacklist"
+
+- name: Leaf - Disable interface again
+  cisco.aci.aci_outofsvc_interface_blacklist:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: present
+  register: disable_int_again
+
+- name: Leaf - Verify that outofsvc_interface_blacklist remains created after disabling again
+  assert:
+    that:
+    - disable_int_again is not changed
+
+- name: Leaf - Query interface
+  cisco.aci.aci_outofsvc_interface_blacklist:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: query
+  register: query_int_disabled
+
+- name: Leaf - Verify that outofsvc_interface_blacklist remains created after query
+  assert:
+    that:
+    - query_int_disabled is not changed
```

Comment:
You should check some value in the return as it could be not changed but empty result.
I would also add a query for all blacklisted interfaces and do a length check.